receiver/receivercreator/runner_test.go

```diff
@@ -69,3 +71,59 @@ func Test_loadAndCreateRuntimeReceiver(t *testing.T) {
 		}())
 	})
 }
+
+func TestValidateSetEndpointFromConfig(t *testing.T) {
```

Comment:
I would consider breaking this test up into a test table so that you can use `assert` instead of `require` to help surface errors within the same test run instead of requiring incremental changes that could conflict with each other.